File: src/main/java/org/alham/alhamfirst/service/orchestrator/OrchestratorTodoService.java
package org.alham.alhamfirst.service.orchestrator;

import org.alham.alhamfirst.document.stat.StatDocument;
import org.alham.alhamfirst.document.stat.UserStatDocument;
import org.alham.alhamfirst.dto.quest.QuestDTO;
import org.alham.alhamfirst.dto.todo.TodoDTO;

import java.util.List;

public interface OrchestratorTodoService {

    public List<QuestDTO> getUnDoQuestListByEncryptedUserId(String encryptedId);
    public void getTodoById(Long todoId);

    public StatDocument createTodo(TodoDTO todoDTO, String encryptedId);

    public UserStatDocument completeTodo(TodoDTO todoDTO);

    public UserStatDocument unCompleteTodo(TodoDTO todoDTO);

    public void deleteTodoById(String todoId);


}

File: src/main/java/org/alham/alhamfirst/service/orchestrator/OrchestratorTodoService.kt
package org.alham.alhamfirst.service.orchestrator;

import org.alham.alhamfirst.domain.document.stat.StatDocument
import org.alham.alhamfirst.domain.dto.quest.QuestDTO
import org.alham.alhamfirst.domain.dto.stat.UserStatDTO
import org.alham.alhamfirst.domain.dto.todo.TodoDTO


interface OrchestratorTodoService {


    fun getUnDoQuestListByEncryptedUserId(encryptedId: String): List<QuestDTO>
    fun getTodoById(todoId: Long)
    fun createQuest(todoDTO: TodoDTO, encryptedId: String): QuestDTO
    fun completeQuest(todoId: Long, encryptedId: String, completed: Boolean): UserStatDTO
    fun deleteTodoById(todoId: String)

    fun changeQuest(todoDTO: TodoDTO, encryptedId: String): QuestDTO
}
